src/main/java/com/aws/emr/profiler/core/requestprocessor/HTTPRequestProcessor.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file.
+ * This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.aws.emr.profiler.core.requestprocessor;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class HTTPRequestProcessor {
+    private static OkHttpClient httpClient = new OkHttpClient();
+    private static String INSTANCE_METADATA_ENDPOINT = "169.254.169.254";
+    private static String INSTANCE_METADATA_URI = "http://%s/latest/meta-data/local-hostname";
```

Comment:
nit: This won't be correct if the cluster is using a custom DNS server as it's provided by ec2 control plane (default VPC DNS server). Might want to use InetAddress or 'ip' commands.

Can take up support for custom dns as a different change

Reply:
Create https://github.com/awslabs/emr-workload-profiler/issues/17 to track this for later